tendrl/monitoring_integration/grafana/alert_dashboard.py

```diff
@@ -104,161 +102,154 @@ def create_resource_dashboard(
-                            resource_name
-                        )
-                        dash_json = create_updated_dashboard(
-                            alert_dashboard, alert_row
-                        )
-                        resp = alert_utils.post_dashboard(dash_json)
-            except Exception as ex:
-                logger.log("error", NS.get("publisher_id", None),
-                           {'message': "Error while updating "
-                            "dashboard for %s" % resource_name})
-                raise ex
-    return resp
+            if sds_name == constants.GLUSTER:
+                alert_rows = fetch_row(alert_dashboard)
+                add_gluster_resource_panel(
+                    alert_rows,
+                    integration_id,
+                    resource_type,
+                    resource_name,
+                    highest_panel_id
+                )
+                alert_dashboard = create_updated_dashboard(
+                    alert_dashboard, alert_rows
+                )
+                highest_panel_id = alert_rows[-1]["panels"][-1]["id"]
+        except Exception as ex:
+            logger.log("error", NS.get("publisher_id", None),
+                       {'message': "Error while updating "
+                        "dashboard for %s" % resource_name})
+            raise ex
+    return alert_dashboard
 
 
 def check_duplicate(
-    alert_dashboard, integration_id, resource_type, resource_name
+    resource_json, resources, resource_type
 ):
-    flag = False
-    for row in alert_dashboard["dashboard"]["rows"]:
-        if "panels" in row and (not flag):
-            for target in row["panels"][0]["targets"]:
-                resource = resource_name
-                if resource_type == "bricks":
-                    hostname = resource.split(":")[0].split(
-                        "|")[1].replace(".", "_")
-                    resource = "." + resource.split(
-                        ":", 1)[1].replace("/", "|") + "."
-                if resource is not None:
-                    if str(integration_id) in target["target"] and str(
-                            resource) in target["target"]:
-                        if resource_type == "bricks":
-                            if hostname in target["target"]:
-                                flag = True
+    # Keeping rows which are match with newly collcted cluster details
+    alert_dashboard = copy.deepcopy(resource_json)
+    resource_json["dashboard"]["rows"] = []
+    new_resource = []
+    highest_panel_id = 1
+    for resource in resources:
+        new_resource_flag = True
+        integration_id = str(resource["integration_id"])
+        for row in alert_dashboard["dashboard"]["rows"]:
+            if "panels" in row:
+                for target in row["panels"][0]["targets"]:
+                    resource_name = str(resource["resource_name"])
+                    if resource_type == "bricks":
```

Comment:
What is the logic of this function. Why only the 'brick' resource type is mentioned here?

Reply:
Here i am identifying whether the panel is already existing for a particular resource. i am identifying by comparing resource name with a target in grafana resource panel json.

for volume, target is like tendrl.clusters.{cid}.olumes.{vol_name}.pcnt_used
so here checking volume name is enough, if it matches then i panel is already exists no need to create new one.

for hosts
tendrl.clusters.{cid}.nodes.{hostname}.memory.percent-buffered 
here comparing with hostname then panel  already  exists

but for brick
tendrl.clusters.{cid}.nodes.{hostname}.bricks.{brick_path}.percent-percent_bytes

so here i need to check both so i put if condition there it find whether type is brick or not, if yes fetch both and compare